### DISCUSSION TITLE
Strict pattern matching mode

### DISCUSSION THREAD

**Discussion by thenonameguy** [noprompt/meander#214] [Ideas]
It would be great to have an `m/find!` macro, which keeps the interface of the bang-less one, but instead of returning `nil` when the passed in value does not match the pattern it returns an exception/error object that contains `why` the match failed.
This would alleviate the current pain of finding the missing/wrong part of the input, without having to re-encode the expected structure in Malli/clojure.spec, etc.

**Comment by noprompt** [comment]
The first half can be implemented as

```clj
(defmacro find! [& rules]
  `(m/find ~@rules _otherwise# (throw (ex-info "Non-exhaustive pattern match" {})))
```

The second part, however, is not something `epsilon` was designed for but is planned for `zeta` (which is still in development).
